epicweb-dev/gratitext
# Update to latest epic stack

This PR contains the latest updates from epicweb-dev/epic-stack

Please review the changes, and feel free to reach out if you encounter any issues.
- /.github/workflows/deploy.yml
  - Remove Dockerfile move step from deploy workflow
- /public/favicons/apple-touch-icon.png 
- /public/favicons/favicon.svg 
- /app/components/error-boundary.tsx
  -  Update arrow function syntax in unexpectedErrorHandler
- /app/components/forms.tsx
  -  Update arrow function syntax for consistency
- /app/components/progress-bar.tsx
  -  Add 'void' keyword to Promise.allSettled call in EpicProgress component
- /app/components/search-bar.tsx
  -  Update arrow function syntax in SearchBar component's onChange handler
- /app/components/ui/input-otp.tsx
  -  Update InputOTPSlot text size for better responsiveness
- /app/components/ui/input.tsx
  -  Update Input component: Adjust text size for responsive design
- /app/components/ui/textarea.tsx
  -  Update Textarea component: Adjust text size for responsive design
- /app/entry.client.tsx
  -  Add 'void' keyword to dynamic import for better error handling
- /app/entry.server.tsx
  -  Update server-side entry point: Improve error handling, remove monitoring init, and update imports
- /app/root.tsx
  -  Update favicon and icon links, add SearchBar and UserDropdown components, and refactor App component layout
- /app/routes/_seo+/sitemap[.]xml.ts
  - Update serverBuild type and access routes through build property
- /app/routes/resources+/healthcheck.tsx
  -  Refactor: Update arrow function syntax in healthcheck route
- /app/routes/resources+/theme-switch.tsx
  -  Add progressive enhancement for theme switching with redirect support
- /app/utils/cache.server.ts 
  - Update cache utility: Use node:fs, add parentheses to arrow functions
- /app/utils/db.server.ts
  - Refactor: Minor syntax improvements in Prisma client setup
- /app/utils/misc.tsx
  -  Refactor: Improve code formatting and arrow function syntax
- /app/utils/misc.use-double-check.test.tsx
  -  Refactor: Improve code formatting in TestComponent
- /app/utils/timing.server.ts
  - Refactor: Update arrow function syntax for consistency
- /app/utils/user.ts
  - Update arrow function syntax in userHasPermission and userHasRole functions
- /fly.toml
  - Add build configuration and adjust internal port
- /index.js
  - Update fs import to use node:fs module
- /other/Dockerfile 
  - Add generation of INTERNAL_COMMAND_TOKEN to Dockerfile
- /other/.dockerignore 
- /other/build-icons.ts
  - Refactor: Update arrow function syntax in build-icons.ts
- /other/build-server.ts
  - Update path and url imports to use node: prefix
- /package.json Update dependencies and configurations in package.json
- /playwright.config.ts
  - Update webServer configuration to always reuse existing server
- /prisma/seed.ts
  - Update catch block syntax in seed.ts
- /public/favicons/README.md
  - Update favicon README: Clarify icon usage for different devices and locations
- /public/favicons/favicon-16x16.png 
- /public/favicons/favicon-32x32.png 
- /public/favicons/mask-icon.svg 
- /server/index.ts
  - Update server configuration and error handling
- /tests/db-utils.ts
  - Refactor cleanupDb function to use try-catch-finally for better error handling and ensure foreign key constraints are always re-enabled
- /tests/playwright-utils.ts
  - Refactor cookie handling and improve type safety in Playwright utils
- /tests/setup/custom-matchers.ts
  - Refactor: Replace custom getSetCookie function with native getSetCookie method
- /tests/setup/setup-test-env.ts
  - Remove @remix-run/node import and installGlobals() call, update consoleError type
- /vite.config.ts 
  - Update Vite config: Add envOnlyMacros, modify external dependencies, expand assetsInlineLimit, conditionally apply remix plugin, and add test configuration
- /vitest.config.ts 

This PR was automatically generated by [Sly which doesn't have a good website yet](https://www.jacobparis.com/).
